### PR TITLE
Upgrade Netty version from 4.1.68 to 4.1.72

### PR DESCRIPTION
**Pull request by Bennett-Lynch**
Upgrade Netty version from 4.1.68 to 4.1.72.

1. https://netty.io/news/2021/10/11/4-1-69-Final.html
2. https://netty.io/news/2021/10/11/4-1-70-Final.html
3. https://netty.io/news/2021/12/09/4-1-71-Final.html
4. https://netty.io/news/2021/12/13/4-1-72-Final.html

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: http-clients/netty-nio-client/src/test/java/software/amazon/awssdk/http/nio/netty/internal/nrs/HandlerSubscriberTest.java [hunk 1]
```diff
@@ -186,6 +194,7 @@ public boolean awaitTermination(long timeout, TimeUnit unit) {
 
         @Override
         public ChannelFuture register(Channel channel) {
+            ((CustomEmbeddedChannel) channel).loop = this;
```
Comment (Bennett-Lynch) [inline http-clients/netty-nio-client/src/test/java/software/amazon/awssdk/http/nio/netty/internal/nrs/HandlerSubscriberTest.java:197]: Needed to support the change in behavior introduced by https://github.com/netty/netty/commit/ef1f637b0f9979d98dbb0f3e5281d7272012c76d:

```
java.lang.ClassCastException: software.amazon.awssdk.http.nio.netty.internal.nrs.HandlerSubscriberTest$CustomEmbeddedEventLoop cannot be cast to io.netty.channel.embedded.EmbeddedEventLoop
--
at software.amazon.awssdk.http.nio.netty.internal.nrs.HandlerSubscriberTest.teardown(HandlerSubscriberTest.java:69)
 
[System Output]
2021-12-15 16:41:35,290 [main] WARN  io.netty.channel.embedded.EmbeddedChannel - More than one exception was raised. Will report only the first one and log others.
java.lang.ClassCastException: software.amazon.awssdk.http.nio.netty.internal.nrs.HandlerSubscriberTest$CustomEmbeddedEventLoop cannot be cast to io.netty.channel.embedded.EmbeddedEventLoop
at io.netty.channel.embedded.EmbeddedChannel.embeddedEventLoop(EmbeddedChannel.java:680)
at io.netty.channel.embedded.EmbeddedChannel.runPendingTasks(EmbeddedChannel.java:602)
at io.netty.channel.embedded.EmbeddedChannel$EmbeddedUnsafe$1.beginRead(EmbeddedChannel.java:842)
at io.netty.channel.DefaultChannelPipeline$HeadContext.read(DefaultChannelPipeline.java:1362)
at io.netty.channel.AbstractChannelHandlerContext.invokeRead(AbstractChannelHandlerContext.java:686)
at io.netty.channel.AbstractChannelHandlerContext.read(AbstractChannelHandlerContext.java:671)
at io.netty.channel.DefaultChannelPipeline.read(DefaultChannelPipeline.java:1004)
at io.netty.channel.AbstractChannel.read(AbstractChannel.java:290)
at io.netty.channel.DefaultChannelPipeline$HeadContext.readIfIsAutoRead(DefaultChannelPipeline.java:1422)
at io.netty.channel.DefaultChannelPipeline$HeadContext.channelActive(DefaultChannelPipeline.java:1400)
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelActive(AbstractChannelHandlerContext.java:230)
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelActive(AbstractChannelHandlerContext.java:216)
at io.netty.channel.DefaultChannelPipeline.fireChannelActive(DefaultChannelPipeline.java:895)
at io.netty.channel.AbstractChannel$AbstractUnsafe.register0(AbstractChannel.java:522)
at io.netty.channel.AbstractChannel$AbstractUnsafe.register(AbstractChannel.java:480)
at io.netty.channel.embedded.EmbeddedChannel$EmbeddedUnsafe$1.register(EmbeddedChannel.java:799)
at software.amazon.awssdk.http.nio.netty.internal.nrs.HandlerSubscriberTest$CustomEmbeddedEventLoop.register(HandlerSubscriberTest.java:195)
at software.amazon.awssdk.http.nio.netty.internal.nrs.HandlerSubscriberTest$CustomEmbeddedEventLoop.register(HandlerSubscriberTest.java:189)
at software.amazon.awssdk.http.nio.netty.internal.nrs.HandlerSubscriberTest.setup(HandlerSubscriberTest.java:61)
```

File: bom-internal/pom.xml [hunk 2]
```diff
@@ -141,9 +141,8 @@
             </dependency>
             <dependency>
                 <groupId>io.netty</groupId>
-                <artifactId>netty-transport-native-epoll</artifactId>
+                <artifactId>netty-transport-classes-epoll</artifactId>
```
Comment (Bennett-Lynch) [inline bom-internal/pom.xml:144]: Motivated by build failure:
```
[INFO] --- maven-dependency-plugin:3.1.1:analyze-only (default) @ netty-nio-client ---
[WARNING] Used undeclared dependencies found:
[WARNING]    io.netty:netty-transport-classes-epoll:jar:4.1.72.Final:compile
[WARNING] Unused declared dependencies found:
[WARNING]    io.netty:netty-transport-native-epoll:jar:linux-x86_64:4.1.72.Final:compile
```

**Review by cenedhryn — APPROVED** [review]

**Comment by lokanx** [comment]
Any plans to merge and release this PR soon?

**Comment by Bennett-Lynch** [comment]
Hi @lokanx. I will try to get this PR merged and released soon. It is currently blocked on some additional testing.

If you have an urgent need to upgrade your consumed Netty version before the SDK change is released, you can also explicitly declare the version of Netty that you would like to use, rather than relying on the SDK's transitive dependency. Minor/patch version increases should generally be backwards compatible. An example of doing that with Maven can be found [here](https://stackoverflow.com/a/3937255) and with Gradle can be found [here](https://docs.gradle.org/current/userguide/dependency_constraints.html).

That said, I understand this may not be a convenient option for all users, and they may still prefer the SDK to pull in the preferred version, so I will continue to prioritize this.

**Comment by Bennett-Lynch** [comment]
Released today as part of version `2.17.107`.

File: bom-internal/pom.xml (hunk 2 above)
Comment (eejbyfeldt) [reply bom-internal/pom.xml:144]: Is the dependency `netty-transport-classes-epoll` really useful without one version of `netty-transport-native-epoll` which contains the native library use by `netty-transport-classes-epoll`? Or are user expected to add their own dependency on `netty-transport-classes-epoll`?

**Pull request merged**